pydantic/env_settings.py

```diff
@@ -56,8 +60,21 @@ def _build_environ(self) -> Dict[str, Optional[str]]:
             d[field.alias] = env_val
         return d
 
+    def _read_from_env_file(self, file_path: str) -> Dict[str, str]:
+        file_vars = {}  # type: Dict[str, str]
+        with open(file_path) as env_file:
+            for line in env_file.readlines():
+                line = line.strip()
+                if '=' in line and not line.startswith('#'):
```

Comment:
```suggestion
                if line.startswith('#'):
                    continue
                elif '=' in line:
                    raise ...
```
better to raise an error if the line does not start with '#' and does not contain '='

Reply:
This might all be easier with a regex something like `re.match('\s*(\w+)\s*=\s*(.+?)\s*', line)` and raise an error if the regex can't be found.

Reply:
I did it this way because it covers all use-cases (AFAIK) and just ignores blank lines, which are fine to have in a `.env` I think.

Reply:
okay, don't mind which.

Reply:
yes, blank lines (or lines with only white space) should be skipped.

Reply:
I'm thinking proper syntax validation of your `.env` file is beyond the scope of pydantic, so would you be fine with just logging a warning if we hit a line that looks bad?

Reply:
If we can't check the format of env files, then we can't accomplish this feature and we should close this PR. But personally I don't think it should be very difficult.

Definitely parsing env files should either succeed or fail, warnings would just make things more complicated.

Reply:
Probably we should just include `python-dotenv`'s implementation of a parser then, no?

https://github.com/theskumar/python-dotenv/blob/master/src/dotenv/parser.py

Reply:
No, it's very simple, please just implement directly as discussed above.